core/src/main/java/com/oath/oak/OakDetachedReadValueBuffer.java

```diff
@@ -6,232 +6,31 @@
-        Slice s = start(getValueSlice());
-        if (index < 0) {
-            throw new IndexOutOfBoundsException();
-        }
-        f = s.getByteBuffer().getFloat(index + valuePosition());
-        end(s);
-        return f;
-    }
-
-    @Override
-    public double getDouble(int index) {
-        double d;
-        Slice s = start(getValueSlice());
-        if (index < 0) {
-            throw new IndexOutOfBoundsException();
-        }
-        d = s.getByteBuffer().getDouble(index + valuePosition());
-        end(s);
-        return d;
-    }
-
-    /**
-     * Returns a transformation of ByteBuffer content.
-     *
-     * @param transformer the function that executes the transformation
-     * @return a transformation of the ByteBuffer content
-     * @throws NullPointerException if the transformer is null
-     */
-    public <T> T transform(OakTransformer<T> transformer) {
-        if (transformer == null) {
-            throw new NullPointerException();
-        }
-        // Use a "for" loop to ensure maximal retries.
-        for (int i = 0; i < 1024; i++) {
-            Result<T> result = valueOperator.transform(getValueSlice(), transformer, version);
-            if (result.operationResult == FALSE) {
-                throw new ConcurrentModificationException();
-            } else if (result.operationResult == RETRY) {
-                lookupValueReference();
-                continue;
-            }
-            return result.value;
-        }
-
-        throw new RuntimeException("Transform failed: reached retry limit (1024).");
-    }
-
-    private Slice start(Slice valueSlice) {
-        ValueUtils.ValueResult res = valueOperator.lockRead(valueSlice, version);
-        if (res == FALSE) {
-            throw new ConcurrentModificationException();
-        }
-        // In case the value moved or was the version does not match
-        if (res == RETRY) {
-            lookupValueReference();
-            return start(getValueSlice());
-        }
-        return valueSlice;
-    }
-
-    private void end(Slice valueSlice) {
-        valueOperator.unlockRead(valueSlice, version);
-    }
-
-    private void lookupValueReference() {
-        EntrySet.LookUp lookUp = internalOakMap.refreshValuePosition(keyReference);
-        if (lookUp == null || lookUp.valueSlice == null) {
-            throw new ConcurrentModificationException();
-        }
-        valueReference = lookUp.valueReference;
-        version = lookUp.version;
-    }
-
-    /*-------------- OakUnsafeRef --------------*/
+/**
+ * This class is used for when a detached access to the value is needed without synchronization.
+ * It does not have to acquire a read lock before each access.
+ * Thus, it can only be used without other concurrent writes in the background.
+ * It is used by stream iterators that iterate over the values (ValueStreamIterator and EntryStreamIterator).
+ */
+class OakDetachedReadValueBuffer extends OakDetachedReadKeyBuffer {
```

Comment:
Why to extend OakDetachedReadKeyBuffer? Why OakDetachedReadBuffer is not enough?